### PR TITLE
[azeventhubs] Bug fixes: send to 'any' partition, also offset position strings were incorrect.

### PR DESCRIPTION
**Pull request by richardpark-msft**
Fixing two bugs and adding in some tests:
- When you send a batch of events you can choose to send it to any available partition. I missed a nil check, added in and added a test.
- The offset position filter strings were being incorrectly formatted. It appears that the sequence number one was accepted _anyways_, but I've now fixed it so they're aligned with what JS does.

### CODE REVIEW

File: sdk/messaging/azeventhubs/consumer_client.go [hunk 1]
```diff
@@ -220,34 +217,66 @@ func (cc *ConsumerClient) Close(ctx context.Context) error {
...
 
-func getOffsetExpression(startPosition StartPosition) string {
+func getOffsetExpression(startPosition StartPosition) (string, error) {
 	lt := ">"
 
 	if startPosition.Inclusive {
 		lt = ">="
 	}
 
+	var errMultipleFieldsSet = errors.New("Only a single start point can be set: Earliest, EnqueuedTime, Latest, Offset, or SequenceNumber")
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/consumer_client.go:227]: Don't capitalize the first letter in error messages (the linter will bark at this).

**Review by jhendrixMSFT — APPROVED** [review]
One little nit to clean up.

File: sdk/messaging/azeventhubs/consumer_client.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/consumer_client.go:227]: Gah, and I _know_ this and still did it.

**Pull request merged**